client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -708,6 +725,14 @@ class CalypsoifyIframe extends Component<
 					source=""
 					visible={ isMediaModalVisible }
 				/>
+				{ hasCheckoutOverlay && (
+					<AsyncLoad
+						require="blocks/editor-checkout-modal"
+						onClose={ this.closeCheckoutModal }
+						isOpen={ isCheckoutModalVisible }
+						cartData={ cartData }
+					/>
+				) }
```

Comment:
I was thinking about this yesterday. This is loaded async indeed. But do we need to load it for everyone who may or may not click "Upgrade"?

I think this is worth a try and see how it feels: 

```jsx
{ hasCheckoutOverlay && isCheckoutModalVisible (
	<AsyncLoad
		require="blocks/editor-checkout-modal"
		onClose={ this.closeCheckoutModal }
		isOpen
		cartData={ cartData }
	/>
) }
```

Reply:
Yup true! This ensure it is only loaded when it really needs to be visible at the trade off of a bit of load time. Let's see how the load time feels like.

Reply:
Same convo here: https://github.com/Automattic/wp-calypso/pull/45932#discussion_r495781898 — Just FYI.